MdeModulePkg/Include/Protocol/NonDiscoverableDeviceUniqueId.h

```diff
@@ -0,0 +1,67 @@
+/** @file NonDiscoverableDeviceUniqueId.h
+  With this new protocol, a platform can register a UniqueId to be used to identify the PciIo location.
+  For example, if a device registers with the NonDiscoverableDeviceRegistrationLib,
+  they can also publish this new protocol on the same EFI_HANDLE as RegisterNonDiscoverableMmioDevice.
+  NonDiscoverablePciDeviceDxe will consume this UniqueId if found and set it to the device's UniqueId.
+  If not found, will back up to the prior method of assigning the UniqueId via a static counter.
+  The platform must maintain all UniqueId's assigned are unique for every instance of the protocol.
+
+  The platform may need to use this new protocol defined UniqueId instead because
+  the current UniqueId that gets used in NonDiscoverablePciDeviceDxe is a static counter,
+  which can lead to non-deterministic value's being returned by PciIo->GetLocation().
+  In order to have a deterministic UniqueId returned by GetLocation() this protocol
+  can be used to specify a specific UniqueId for a given Handle.

```

Comment:
```suggestion
  Platforms may need to define their own UniqueIds through this method because
  the current UniqueId that gets used in NonDiscoverablePciDeviceDxe is a static counter,
  which can lead to non-deterministic value's being returned by PciIo->GetLocation().
  In order to have a deterministic UniqueId returned by GetLocation() this protocol
  can be used to specify a specific UniqueId for a given Handle.
```

I think adding another sentence here to give an example scenario where you'd want a deterministic UniqueId would be useful.

Reply:
done, added:
For example, if a platform needs to Locate a specific NonDiscoverable PciIo protocol,
they could use the deterministic UniqueId assigned through this protocol to find the expected PciIo instance
because a platform can match against the GetLocation value returned by using the assigned UniqueId.
This protocol allows us to have a 1-1 defined mapping between a platform defined UniqueId and the NonDiscoverable PciIo protocol instance.